tutorials/cloud-iot-rtdp/index.md

```diff
@@ -0,0 +1,100 @@
+---
+title: Real time Data Processing With Google Cloud IoT
+description: Demonstrate how to build data processing backend for Google Cloud IoT.
+author: teppeiy
+tags: Cloud IoT Core, Cloud Dataflow, Cloud Functions, Java, Node.js
+date_published: 2017-11-29
+---
+# Real time Data Processing With Google Cloud IoT
+This repository contains sample code for Real time Data Processing With Google Cloud IoT solution. Sample MQTT client generates simulated data and submits them to Cloud IoT Core. Cloud Functions and Cloud Dataflow receive data through Cloud Pub/Sub.
+
+![Architecture](arch.png)
+
+## Prerequisites
+- Git
+- Google Cloud SDK
+- Java 1.8 or later
+- Apache Maven 3.3.x or later
+
+All commands in following steps are assumed to be executed under "community/tutorials/cloud-iot-rtdp" directory.
+
+## Enable APIs
+Enable following APIs from Google Cloud Console.
+- Cloud IoT Core
+- Cloud Functions
+- Cloud Dataflow
+
+## Create a Cloud Storage bucket and set environment variables.
+1. Create a Cloud Storage bucket and a temporary folder.
+2. Set environment variables. Corresponding resources are created in later steps.
+```bash
+% export PROJECT=<GCP Project ID>
+% export BUCKET=<Cloud Storage Bucket Name>
+% export REGISTRY=<Cloud IoT Core Registry ID>
+% export TOPIC=<Cloud Pub/Sub Topic Name>
+```
+
+## Configure Cloud IoT Core
+1. Open Cloud Pub/Sub console and create a new topic.
+2. Move to Cloud IoT Core console and click create device registry.
+3. Enter Registry ID, select a region and choose the Pub/Sub topic created in step 1.
+4. Execute a script to register devices in the device registry.
+```bash
+% bin/register.sh
+```
+
+## Deploy a function to Cloud Functions
+Execute following commands to deploy a function to Cloud Functions.
+
+```bash
+% cd function
+
+% gcloud beta functions deploy iot --stage-bucket $BUCKET --trigger-topic $TOPIC
+/ [1 files][  292.0 B/  292.0 B]
+Operation completed over 1 objects/292.0 B.
+Deploying function (may take a while - up to 2 minutes)...done.
+availableMemoryMb: 256
+entryPoint: iot
+eventTrigger:
+```
+
+## Deploy a streaming application to Cloud Dataflow
+Build and submit a streaming job by executing following commands.
+
+```bash
+% cd bin
+
+% ./job.sh
+[INFO] Scanning for projects...
+[INFO]                                                                        
+...
+```
+
+## Generate simulated temperature and coordinates data
+Execute an MQTT client to generate simulated temperature and coordinates data.
```

Comment:
This is the first time you mention the temperature thing. Please add some explanation in the introduction.